ktor-core-tests/test/org/jetbrains/ktor/tests/session/SessionTest.kt

```diff
@@ -479,13 +487,21 @@ class SessionTest {
             }
 
             handleRequest(HttpMethod.Get, "/2") {
-                val brokenSession = sessionId.mapIndexed { i, c -> if (i == sessionId.lastIndex) 'x' else c }.joinToString("")
+                val brokenSession = flipLastHexDigit(sessionId)
                 addHeader(cookieName, brokenSession)
             }.let { call ->
                 assertEquals("ok, null, false", call.response.content)
             }
         }
     }
+
+    private fun flipLastHexDigit(sessionId: String) = sessionId.mapIndexed { index, letter ->
```

Comment:
'x' is not a valid HEX digit -> NumberFormatException later, when hex() is called
Still, one old test for values with 'x' is preserved